src/server/models/user/router.js

```diff
@@ -0,0 +1,65 @@
+const router = require('express').Router();
+const User = require('./model');
+
+const passport = require('passport');
+const Strategy = require('passport-local').Strategy;
+
+passport.use(new Strategy(
+    function(username, password, done) {
+        User.findOne({ username })
+            .then((user) => {
+                if (!user || password !== user.password) {
+                    return done(null, false);
+                }
+                done(null, user)
+            })
+            .catch(done)
+    }));
+
+passport.serializeUser(function(user, done) {
+    done(null, user._id);
+});
+    
+passport.deserializeUser(function(id, done) {
+    User.findById(id)
+        .then((user) => {
+            done(null, user)
+        })
+        .catch(done)
+});
+
+router.route('/api/login')
+    .post(
+        passport.authenticate('local'),
+        (req, res) => {
+            res.status(200).json({userName: req.user.username});
+        }
+    );
+
+router.route('/api/logout')
+    .get((req, res) => {
+        req.logout();
+        res.sendStatus(200);
+    });
+
+router.route('/api/registration')
+    .post(async (req, res, next) => {
```

Comment:
async is redundant because you have used promise without await

Reply:
fixed